src/Middlewares/PermissionMiddleware.php

```diff
@@ -9,7 +9,13 @@ class PermissionMiddleware
 {
     public function handle($request, Closure $next, $permission)
     {
+        $routeNameRedirect = config('permission.unauthorized_route_name_redirect');
+
         if (Auth::guest()) {
+            if (! is_null($routeNameRedirect)) {
+                return redirect()
+                    ->route($routeNameRedirect);
+            }
             abort(403);
```

Comment:
These 6 lines of code are being used in 4 different places. Can we put them somewhere in their own method? `handleUnauthorized()` or something similar. Maybe use a trait that can be used in both `PermissionMiddleware` and `RoleMiddleware`?

Reply:
@AlexVanderbist , yes much better that, i will push another commit later :)